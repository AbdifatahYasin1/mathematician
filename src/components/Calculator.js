import React, { useState } from 'react';
import calculate from '../logic/calculate';

function Calculator() {
  const intialData = { total: 0, next: null, operation: null };
  const [input, setInput] = useState(intialData);

  const eventHandler = (e) => {
    const btnClicked = e.target.innerText;
    const result = calculate(input, btnClicked);
    setInput(result);
  };
  const { total, operation, next } = input;

  return (

    <div className="container">
      <div className="sero">
        {total}
        {operation}
        {next}
      </div>
      <button type="button" className="border gray" onClick={eventHandler}>AC</button>
      <button type="button" className="border " onClick={eventHandler}> &#43;/&minus; </button>
      <button type="button" className="border " onClick={eventHandler}>%</button>
      <button type="button" className="border orange " onClick={eventHandler}>&divide;</button>

      <button type="button" className="border gray" onClick={eventHandler}>7</button>
      <button type="button" className="border " onClick={eventHandler}> 8 </button>
      <button type="button" className="border gray" onClick={eventHandler}>9</button>
      <button type="button" className="border orange" onClick={eventHandler}>&times;</button>

      <button type="button" className="border gray" onClick={eventHandler}>4</button>
      <button type="button" className="border " onClick={eventHandler}> 5 </button>
      <button type="button" className="border gray" onClick={eventHandler}>6</button>
      <button type="button" className="border orange " onClick={eventHandler}>-</button>

      <button type="button" className="border gray" onClick={eventHandler}>1</button>
      <button type="button" className="border " onClick={eventHandler}>2 </button>
      <button type="button" className="border gray" onClick={eventHandler}>3</button>
      <button type="button" className="border orange " onClick={eventHandler}>+</button>

      <button type="button" className="border span " onClick={eventHandler}>0 </button>
      <button type="button" className="border gray" onClick={eventHandler}>.</button>
      <button type="button" className="border " onClick={eventHandler}>=</button>

    </div>

  );
}

export default Calculator;
