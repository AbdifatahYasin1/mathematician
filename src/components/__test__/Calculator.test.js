import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('Testing Calculator', () => {
  test('should render correctly', () => {
    render(<Calculator />);
  });
});

test('check textcontent of buttons', () => {
  const { getAllByRole } = render(<Calculator />);
  const buttons = getAllByRole('button');
  expect(buttons[0].textContent).toBe('AC');
});

it('fisrt snap shot test', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
