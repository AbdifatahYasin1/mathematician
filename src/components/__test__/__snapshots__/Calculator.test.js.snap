// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fisrt snap shot test 1`] = `
<div
  className="container"
>
  <div
    className="sero"
  >
    0
  </div>
  <button
    className="border gray"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="border "
    onClick={[Function]}
    type="button"
  >
     +/− 
  </button>
  <button
    className="border "
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="border orange "
    onClick={[Function]}
    type="button"
  >
    ÷
  </button>
  <button
    className="border gray"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="border "
    onClick={[Function]}
    type="button"
  >
     8 
  </button>
  <button
    className="border gray"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="border orange"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="border gray"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="border "
    onClick={[Function]}
    type="button"
  >
     5 
  </button>
  <button
    className="border gray"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="border orange "
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="border gray"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="border "
    onClick={[Function]}
    type="button"
  >
    2 
  </button>
  <button
    className="border gray"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="border orange "
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="border span "
    onClick={[Function]}
    type="button"
  >
    0 
  </button>
  <button
    className="border gray"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="border "
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
